### DISCUSSION TITLE
NextJs SSR hydration issue when slot content is moved in firstUpdated

### DISCUSSION THREAD

**Discussion by arunks5** [lit/lit#4581] [General]
Hey, we are using a custom dropdown lit component in nextJS. We will be providing the data to dropdown in default slot. This default slot contents will be appended to body when dropdown is opened to overcome [stacking context](https://developer.mozilla.org/en-US/docs/Web/CSS/CSS_positioned_layout/Understanding_z-index/Stacking_context) issue. This works fine as expected.
But when the users want the dropdown to open on load itself, we will be doing the same dom manipulation on firstUpdated and it is throwing hydration errors.
<img width="803" alt="image" src="https://github.com/lit/lit/assets/161848096/97daf38e-cce4-4a17-ad74-d72a4fe739ba">

We are using custom-wrapper to pass the attribute values to the server ----- https://github.com/lit/lit/issues/3418 
I have recreated the scenario in [stackblitz](https://stackblitz.com/edit/nextjs-ljzqjq?file=README.md).

Any help will be appreciated. :)

**Comment by augustjk** [comment]
That's a tricky situation. `firstUpdated()` is running before Next.js has done hydration on the React tree which is why the hydration error is happening. You probably don't want to do any DOM mutation in `firstUpdated()` to avoid this.

One alternative I can think of is to make the moving of the node happen with a method, or some reactive property, and trigger that from the React side with an on mount effect. e.g. https://stackblitz.com/edit/nextjs-nvhmej?file=app%2Fpage.tsx,lib%2Flit%2Fdropdown.ts,lib%2Freact%2Fdropdown.tsx
